react/components/Variation/Items/MultipleChoice.js

```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import NumericStepper from 'vtex.styleguide/NumericStepper'
+import ProductPrice from 'vtex.store-components/ProductPrice'
+
+class MultipleChoice extends Component {
+  static propTypes = {
+    /* Index to define the chossed amount */
+    index: PropTypes.string,
+    /* Object that handle the choosed amount of variations */
+    choosedAmount: PropTypes.object,
+    /* Min limit of selections */
+    minTotalItems: PropTypes.string,
+    /* Max limit of selections */
+    maxTotalItems: PropTypes.string,
+    /* Trigger function to handle changes on inputs  */
+    onVariationChange: PropTypes.func,
+    /* Item to populate component data   */
+    item: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
+  }
+
+  handleChoosedAmount = async e => {
+    e.preventDefault()
+
+    const choosedAmount = e.value
+    const {
+      item,
+      index,
+      minTotalItems,
+      maxTotalItems,
+      onVariationChange,
+    } = this.props
+
+    await this.setState({ choosedAmount })
```

Comment:
Remove await keyword (and the async from the function) and add the call bellow to the second argument of `this.setState` like

```suggestion
    this.setState({ choosedAmount }, () => onVariationChange({ index, minTotalItems, maxTotalItems, variation: item, quantity: choosedAmount }))
```